realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionSnapshot.java

```diff
@@ -0,0 +1,191 @@
+     *
+     * @throws UnsupportedOperationException
+     */
+    @Override
+    public RealmResults<E> sort(String fieldName) {
+        throw getUnsupportedException("sort");
+    }
+
+    /**
+     * Not supported by {@link OrderedRealmCollectionSnapshot}. Use 'sort()' on the original
+     * {@link OrderedRealmCollection} instead.
+     *
+     * @throws UnsupportedOperationException
+     */
+    @Override
+    public RealmResults<E> sort(String fieldName, Sort sortOrder) {
+        throw getUnsupportedException("sort");
+    }
+
+    /**
+     * Not supported by {@link OrderedRealmCollectionSnapshot}. Use 'sort()' on the original
+     * {@link OrderedRealmCollection} instead.
+     *
+     * @throws UnsupportedOperationException
+     */
+    @Override
+    public RealmResults<E> sort(String fieldName1, Sort sortOrder1, String fieldName2, Sort sortOrder2) {
+        throw getUnsupportedException("sort");
+    }
+
+    @Override
+    public RealmResults<E> sort(String[] fieldNames, Sort[] sortOrders) {
+        throw getUnsupportedException("sort");
+    }
+
+    /**
+     * Not supported by {@link OrderedRealmCollectionSnapshot}. Use 'where()' on the original
+     * {@link OrderedRealmCollection} instead.
+     *
+     * @throws UnsupportedOperationException
+     */
+    @Deprecated
+    @Override
+    public RealmQuery<E> where() {
+        throw getUnsupportedException("where");
+    }
+
+    private UnsupportedOperationException getUnsupportedException(String methodName) {
+        return new UnsupportedOperationException(
+                String.format("'%s()' is not supported by OrderedRealmCollectionSnapshot. " +
+                        "Call '%s()' on the original 'RealmCollection' instead.", methodName, methodName));
+    }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isLoaded() {
+        return true;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean load() {
+        return true;
+    }
+
+    @Override
+    public OrderedRealmCollectionSnapshot<E> createSnapshot() {
+        realm.checkIfValid();
+        return this;
+    }
+
+    /**
+     * Deletes the object at the given index from the Realm. The object at the given index becomes invalid. Just returns
+     * if the object is invalid already.
+     *
+     * @param location the array index identifying the object to be removed.
+     * @throws IndexOutOfBoundsException if {@code location < 0 || location >= size()}.
+     * @throws java.lang.IllegalStateException if the Realm is closed or the method is called from the wrong thread.
+     */
+    @Override
+    public void deleteFromRealm(int location) {
+        realm.checkIfValidAndInTransaction();
+        UncheckedRow row = collection.getUncheckedRow(location);
+        if (row.isAttached()) {
+            collection.delete(location);
+        }
+    }
+
+    /**
+     * Deletes the first object from the Realm. The first object becomes invalid.
```

Comment:
The first object will become invalid?

Same above. I don't know whether `The first object becomes invalid.` is suitable or not.